site/src/pages/TerminalPage/TerminalPage.tsx

```diff
@@ -328,6 +328,73 @@ const TerminalPage: FC = () => {
   );
 };
 
+type TerminalAlertsProps = {
+  agent: WorkspaceAgent | undefined;
+  state: TerminalState;
+  onAlertChange: () => void;
+};
+
+const TerminalAlerts = ({
+  agent,
+  state,
+  onAlertChange,
+}: TerminalAlertsProps) => {
+  const lifecycleState = agent?.lifecycle_state;
+  const prevLifecycleState = useRef(lifecycleState);
+  useEffect(() => {
+    prevLifecycleState.current = lifecycleState;
+  }, [lifecycleState]);
+
+  // Mock state
+  const [displayAlert, setDisplayAlert] = useState(false);
+  useEffect(() => {
+    setTimeout(() => {
+      setDisplayAlert(true);
+    }, 1_000);
+  }, []);
+
+  // We want to observe the children of the wrapper to detect when the alert
+  // changes. So the terminal page can resize itself.
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    if (!wrapperRef.current) {
+      return;
+    }
+    const observer = new MutationObserver(onAlertChange);
+    observer.observe(wrapperRef.current, { childList: true });
+
+    return () => {
+      observer.disconnect();
+    };
+  }, [onAlertChange]);
```

Comment:
Would it be possible to just always call `fit()` when this component re-renders instead of using an observer?  Since ideally the component only renders again when the alert changes.

Reply:
This is a good question and the why this does not work is that the `.fit()` needs to run after the render so in this case, I just think the mutation observer is more reliable. I could use some hacky `setTimeout` inside of `useEffect` to do that, I guess, but I don't think it would be any better. Wdyt?

Reply:
PS: I'm going to add this as a comment so we remember the "why" of this decision.

Reply:
Ohhh right that makes sense!  Yeah you are right, this is better.